resources/windows/wix/customactions.wxs.erb

```diff
@@ -198,7 +198,7 @@ End If
     <!-- Due to PUP-6729, system may not have permission to modify DACL, so first take ownership -->
     <SetProperty
         Id="TakeownPuppetPermissions"
-        Value="&quot;[%WINDIR]\System32\takeown.exe&quot; /F &quot;[$PuppetAppDir]\*&quot; /R /A /D N"
+        Value="&quot;[%WINDIR]\System32\takeown.exe&quot; /F &quot;[PuppetAppData]\*&quot; /R /A /D N"
```

Comment:
This still does not work properly, under specific circumstances, due to as of yet unknown circumstances.

It will however work with:

```
Value="&quot;[%WINDIR]\System32\takeown.exe&quot; /F &quot;[%PROGRAMDATA]\PuppetLabs\puppet\*&quot; /R /A /D N"
```

Reply:
During a repair, I see the following, which I assume is what they mean about "the action state of the component is null":

```
MSI (s) (04:98) [03:24:27:695]: Component: PuppetAppDir; Installed: Local;   Request: Null;   Action: Null
```